lib/streamlit/elements/widgets/button_group.py

```diff
@@ -379,30 +543,64 @@ def _button_group(
         selection_visualization: ButtonGroupProto.SelectionVisualization.ValueType = (
             ButtonGroupProto.SelectionVisualization.ONLY_SELECTED
         ),
-        after_register_callback: Callable[
-            [RegisterWidgetResult[T]], RegisterWidgetResult[T]
-        ]
-        | None = None,
+        label: str | None = None,
+        label_visibility: LabelVisibility = "visible",
+        help: str | None = None,
     ) -> RegisterWidgetResult[T]:
+        _maybe_raise_selection_mode_warning(selection_mode)
+
+        parsed_selection_mode: ButtonGroupProto.ClickMode.ValueType = (
+            ButtonGroupProto.SINGLE_SELECT
+            if selection_mode == "single"
+            else ButtonGroupProto.MULTI_SELECT
+        )
+
+        # when selection mode is a single-value selection, the default must be a single
+        # value too.
+        if (
+            parsed_selection_mode == ButtonGroupProto.SINGLE_SELECT
+            and default is not None
+            and isinstance(default, Sequence)
+            and len(default) > 1
+        ):
+            raise StreamlitAPIException(
+                "The default argument to `st.button_group` must be a single value when "
```

Comment:
nit: can this exception only appear on button group usage or also pills / segments / feedback? If so, maybe we write the comment a bit more generic since the `st.button_group` could be confusing otherwise.

Reply:
yeah good point 👍 I think it should be fine to list the commands here (starting with `st.pills`)